pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -252,22 +265,222 @@ func (pl *dynamicResources) EventsToRegister() []framework.ClusterEvent {
 
 	events := []framework.ClusterEvent{
 		// Allocation is tracked in ResourceClaims, so any changes may make the pods schedulable.
-		{Resource: framework.ResourceClaim, ActionType: framework.Add | framework.Update},
+		{Resource: framework.ResourceClaim, ActionType: framework.Add | framework.Update, IsSchedulable: pl.isSchedulableAfterClaimAddOrUpdate},
+
 		// When a driver has provided additional information, a pod waiting for that information
 		// may be schedulable.
-		// TODO (#113702): can we change this so that such an event does not trigger *all* pods?
-		// Yes: https://github.com/kubernetes/kubernetes/blob/abcbaed0784baf5ed2382aae9705a8918f2daa18/pkg/scheduler/eventhandlers.go#L70
-		{Resource: framework.PodSchedulingContext, ActionType: framework.Add | framework.Update},
+		{Resource: framework.PodSchedulingContext, ActionType: framework.Update | framework.Delete, IsSchedulable: pl.isSchedulableAfterPodSchedulingUpdateOrDelete},
+
 		// A resource might depend on node labels for topology filtering.
 		// A new or updated node may make pods schedulable.
 		{Resource: framework.Node, ActionType: framework.Add | framework.UpdateNodeLabel},
 	}
 	return events
 }
 
+// PreEnqueue checks if there are known reasons why a pod currently cannot be
+// scheduled. When this fails, one of the registered events can trigger another
+// attempt.
+func (pl *dynamicResources) PreEnqueue(ctx context.Context, pod *v1.Pod) (status *framework.Status) {
+	logger := klog.FromContext(ctx)
+	defer func() {
+		logger.V(3).Info("PreEnqueue", "pod", klog.KObj(pod), "status", status)
+	}()
+	if err := pl.foreachPodResourceClaim(pod, nil); err != nil {
+		return statusUnschedulable(logger, err.Error())
+	}
+	return nil
+}
+
+// isSchedulableAfterClaimAddOrUpdate is invoked whenever a claim changed. It checks whether
+// that change made a previously unschedulable pod schedulable. It errs on the side of
+// letting a pod scheduling attempt happen.
+func (pl *dynamicResources) isSchedulableAfterClaimAddOrUpdate(oldObj, newObj interface{}, pod *v1.Pod) framework.IsSchedulableResult {
```

Comment:
how the lapsed time looks like in your testing? this is also one concern point when brainstorming the design in https://docs.google.com/document/d/1S1_OqwCN2oY0Xp4dZt6XLCQQGu7dWm96blgXmGKhwSo/edit

Reply:
What is "lapsed time" and how can I measure it? `scheduler_perf` showed no significant difference with the modified framework. Using the filter in the plugin made DRA better without affecting the rest.